src/main/Main.hs

```diff
@@ -817,20 +820,23 @@ uploadCmd (args, mpvpBounds) go = do
             let uploadSettings =
                     Upload.setGetManager (return manager) Upload.defaultUploadSettings
             liftIO $ Upload.mkUploader config uploadSettings
-    if null dirs
-        then withConfigAndLock go $ do
-            uploader <- getUploader
-            liftIO $ forM_ files (canonicalizePath >=> Upload.upload uploader)
-        else withBuildConfigAndLock go $ \_ -> do
-            uploader <- getUploader
-            liftIO $ forM_ files (canonicalizePath >=> Upload.upload uploader)
-            forM_ dirs $ \dir -> do
-                pkgDir <- parseAbsDir =<< liftIO (canonicalizePath dir)
-                (tarName, tarBytes) <- getSDistTarball mpvpBounds pkgDir
-                liftIO $ Upload.uploadBytes uploader tarName tarBytes
+    withBuildConfigAndLock go $ \_ -> do
+      uploader <- getUploader
+      unless ignoreCheck $
+        mapM_ (liftIO . canonicalizePath >=>
+               parseAbsFile              >=>
```

Comment:
`liftIO . canonicalizePath >=> parseAbsFile` is the same as `Path.IO.parseRelAsAbsDir`